modules/agar/src/main/ts/ephox/agar/assertions/ApproxStructures.ts

```diff
@@ -126,6 +130,33 @@ const element = (tag: string, fields: ElementFields): StructAssert => {
   };
 };
 
+const elementWithSpecifics = (tag: string, fields: ElementWithSpecificFields): StructAssert => {
```

Comment:
The name `ElementWithSpecifics` is vague, maybe a name reflecting that it exactly matches classes would be better.

Reply:
Not sure if `elementWithExactMatch`, and the rest `assertExactMatchClasses` sounds better?

Reply:
It also matches styles and attributes. Maybe using `exact` is better, as you say, @shanmen-tiny

Reply:
Could it be separate properties instead for the normal element. Like `exactAttributes`, `exactClasses` then just error out of both `attributes` and `exactClasses` are defined. The type system could have both one with and one without so might not need proper validation?

Reply:
Or maybe it's fine to mix them having like `element('div', { exactClasses: [ str.is('foo') ], attributes: { bar: str.is('bar') } } )`

Reply:
That's a good point that you might want to mix them. I'm wary of using runtime errors though to avoid defining both, though, so there isn't really a good solution for that. There's arguments both ways. I like the separate API approach, but I didn't think that you might want to mix and match them in the same element. I think the key thing that I care about is the decision has to be intentional. I don't want something that just detects the type you use, and acts accordingly, because it will be too easy to write the wrong thing.

Reply:
I'd agree with Spocke here that having it be part of the existing API with specific keys is probably the better way here. While you might accidentally combine `classes` with `exactClasses` we basically already have that problem today anyways with `attributes` and `classes`/`styles` as you can mix and match them.

Reply:
Fair enough. You've both convinced me. Sorry @shanmen-tiny , I'll make some alterations to the ticket. Most of your approach is fine ... it's just that it will be all be part of `s.element` instead of a separate API. It also avoids having to come up with better names as well, as `exactClasses` and `exactStyles` etc. are pretty self-explanatory.

For now (unless anyone strongly objects, we won't have a runtime error if both variants are provided, but the *exact* variants will take priority.

Reply:
I've updated the ticket.

Reply:
Thank you all for the suggestions. I have applied the changes of having the ability to mix and match different variants.